### PR TITLE
[main] [core] Remove in-spec source checksum definitions

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

We inherited a lot of `%define sha1 <hash>` lines in our specs from Photon. In the Mariner build system, these declarations are vestigial since we store source checksums in a separate signatures file. So, let's get rid of the remainder of these lines!

###### Change Log  

- Remove in-spec source checksums in atftp, cppunit, expect, gnuplot, gperf, hdparm, i2c-tools, libevent, libgssglue, libnetfilter_cthelper, libnetfilter_cttimeout, libnfnetlink, libssh2, libtool, lshw, mlocate, perl-DBD-SQLite, python-setuptools_scm, tmux, tree, which, xmlto

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- N/A, low risk change due to the nature of the changes within each package. Toolchain changes should be sufficiently validated by the toolchain manifest check

### CODE REVIEW

File: toolkit/resources/manifests/package/toolchain_aarch64.txt [hunk 1]
```diff
@@ -166,8 +166,8 @@ libgpg-error-lang-1.43-1.cm2.aarch64.rpm
 libksba-1.6.0-1.cm2.aarch64.rpm
 libksba-debuginfo-1.6.0-1.cm2.aarch64.rpm
 libksba-devel-1.6.0-1.cm2.aarch64.rpm
-libltdl-2.4.6-7.cm2.aarch64.rpm
-libltdl-devel-2.4.6-7.cm2.aarch64.rpm
+libltdl-2.4.6-8.cm2.aarch64.rpm
```
Comment (mfrw) [inline toolkit/resources/manifests/package/toolchain_aarch64.txt:169]: I think only this is an entry which is not apparent, but rest seems to checkout .. +1

**Review by mfrw — APPROVED** [review]
LGTM

File: toolkit/resources/manifests/package/pkggen_core_aarch64.txt [hunk 2]
```diff
@@ -74,7 +74,7 @@ xz-libs-5.2.5-1.cm2.aarch64.rpm
 zstd-1.5.0-1.cm2.aarch64.rpm
 zstd-devel-1.5.0-1.cm2.aarch64.rpm
 zstd-libs-1.5.0-1.cm2.aarch64.rpm
-libtool-2.4.6-7.cm2.aarch64.rpm
+libtool-2.4.6-8.cm2.aarch64.rpm
```
[the remark this answers is not in the document]
Comment (thcrain-msft) [reply toolkit/resources/manifests/package/pkggen_core_aarch64.txt:77]: They were automatically checked and confirmed to be valid by the toolchain pipeline build I ran overnight

[the remark this answers is not in the document]
Comment (mfrw) [reply toolkit/resources/manifests/package/pkggen_core_aarch64.txt:77]: :shipit:

**Review by PawelWMS — APPROVED** [review]
I love it that you did that. If you're looking for a quick way to download and extract the sources now to check the licenses, try using one of the dev scripts:

```bash
toolkit/scripts/download_spec_sources.sh cgmanifest.json [space-separated list of spec names]
```

**Comment by thcrain-msft** [comment]
> I love it that you did that. If you're looking for a quick way to download and extract the sources now to check the licenses, try using one of the dev scripts:
> 
> ```shell
> toolkit/scripts/download_spec_sources.sh cgmanifest.json [space-separated list of spec names]
> ```

Neat script! Worked great and saved me some time, thanks!

**Pull request merged**